usaspending_api/agency/v2/views/agency_base.py

```diff
@@ -44,3 +51,26 @@ def toptier_agency(self):
     @property
     def standard_response_messages(self):
         return [get_account_data_time_period_message()] if self.fiscal_year < 2017 else []
+
+    @cached_property
+    def pagination(self):
+        if self.request.method == "GET":
```

Comment:
This seems unnecessary.  The only way you can get to this method is if you actually call it.  I assume you wouldn't call it if your endpoint didn't support it.  Plus, I could see this being useful regardless of whether or not we end up with `POST` or `GET` requests.  (See another comment I'm about to make concerning combining `request.data` with `request.query_params`.)